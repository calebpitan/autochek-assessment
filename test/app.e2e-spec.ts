import { Test, TestingModule } from '@nestjs/testing'
import { INestApplication } from '@nestjs/common'
import request from 'supertest'
import { AppModule } from './../src/app.module'

describe('AppController (e2e)', () => {
  const locationDtoMock = {
    location_name: 'Lagos',
    description: 'Lagos in Nigeria',
    website: 'https://www.lagos.ng',
    phone: '07066001122',
    contact_person: 'Jamie Foxx',
    coordinates: [6.4550575, 3.3941795],
  }

  let app: INestApplication

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile()

    app = moduleFixture.createNestApplication()
    await app.init()
  })

  it('/location (POST)', async () => {
    const req = await request(app.getHttpServer()).post('/location').send(locationDtoMock)

    expect(req.statusCode).toStrictEqual(201)
    expect(req.body).toMatchSnapshot()
  })

  it('/location (GET)', async () => {
    const req = await request(app.getHttpServer()).get('/location')
    expect(req.statusCode).toStrictEqual(200)
    expect(req.body).toMatchSnapshot()
  })

  it('/location/:id (GET)', async () => {
    const req = await request(app.getHttpServer()).get('/location/1')
    expect(req.statusCode).toStrictEqual(200)
    expect(req.body).toMatchSnapshot()
  })

  it('/location/1 (PUT)', async () => {
    const req = await request(app.getHttpServer())
      .put('/location/1')
      .send({ ...locationDtoMock, contact_person: 'Jide Sanwolu' })

    expect(req.statusCode).toStrictEqual(200)
    expect(req.body).toMatchSnapshot()
  })

  it('/location/1 (DELETE)', async () => {
    const req = await request(app.getHttpServer()).delete('/location/1')

    expect(req.statusCode).toStrictEqual(200)
    expect(req.body).toMatchSnapshot()
  })
})
