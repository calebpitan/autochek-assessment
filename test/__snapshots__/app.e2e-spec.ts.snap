// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) /location (POST) 1`] = `
Object {
  "data": Object {
    "location": Object {
      "contact_person": "Jamie Foxx",
      "coordinates": Object {
        "coordinates": Array [
          6.4550575,
          3.3941795,
        ],
        "type": "Point",
      },
      "description": "Lagos in Nigeria",
      "id": 2,
      "location_name": "Lagos",
      "phone": "07066001122",
      "website": "https://www.lagos.ng",
    },
  },
  "message": "Location created successfully!",
  "success": true,
}
`;

exports[`AppController (e2e) /location/:id (GET) 1`] = `
Object {
  "data": Object {
    "location": Object {
      "contact_person": "Caleb Adepitan",
      "coordinates": Object {
        "coordinates": Array [
          6.4550575,
          3.3941795,
        ],
        "type": "Point",
      },
      "description": "Lagos in Nigeria",
      "id": 1,
      "location_name": "Lagos",
      "phone": "07066029547",
      "website": "https://www.lagos.ng",
    },
  },
}
`;

exports[`AppController (e2e) /location/1 (DELETE) 1`] = `
Object {
  "data": Object {
    "affected": 1,
    "raw": Array [],
  },
}
`;

exports[`AppController (e2e) /location/1 (PUT) 1`] = `
Object {
  "data": Object {
    "location": Object {
      "affected": 1,
      "generatedMaps": Array [],
      "raw": Array [],
    },
  },
  "message": "Location updated successfully!",
  "success": true,
}
`;
